### PR TITLE
Enable Zcmt support in core-v-verif tests

### PR DESCRIPTION
**Pull request by silabs-hfegran**
This PR introduces zcmt to riscv-dv randomtesting and enables zcmt compiler flags for all? relevant configurations. 

Note: This _requires_ updated compiler support, tested with: corev-openhw-gcc-centos7-20230310

### CODE REVIEW

**Comment by silabs-hfegran** [comment]
I left a DO NOT MERGE label on this to let the relevant parties get a chance to update their compiler suites before we merge this in, otherwise it should be ready.

**Comment by silabs-hfegran** [comment]
@eroom1966: Are you guys up to speed with the latest compiler as mentioned above?

File: cv32e40s/env/corev-dv/cv32e40s_zcmt_instr_lib.sv
Comment (silabs-robin) [inline cv32e40s/env/corev-dv/cv32e40s_zcmt_instr_lib.sv:150]: Loads _could_ in principle be used, but it is unreasonably complicated?

File: cv32e40s/env/corev-dv/cv32e40s_zcmt_instr_lib.sv
Comment (silabs-robin) [inline cv32e40s/env/corev-dv/cv32e40s_zcmt_instr_lib.sv:152]: What would happen if we unconditionally said `!(rd inside {cfg.reserved_regs})`?

File: cv32e40s/env/corev-dv/cv32e40s_zcmt_instr_lib.sv
Comment (silabs-robin) [inline cv32e40s/env/corev-dv/cv32e40s_zcmt_instr_lib.sv:153]: Does this cover the case where compressed instructions have "rd/rs1" fields?

**Review by silabs-robin — APPROVED** [review]
Approving based on:
A zcmt update was expected, as zcmt is the subextension that was previously not complete.
March looks good.
Enabling the tests looks good.
I only have minor questions about the instr lib.

File: cv32e40s/env/corev-dv/cv32e40s_zcmt_instr_lib.sv
Comment (silabs-hfegran) [reply cv32e40s/env/corev-dv/cv32e40s_zcmt_instr_lib.sv:152]: randomization would fail for instructions that do not have rd

File: cv32e40s/env/corev-dv/cv32e40s_zcmt_instr_lib.sv
Comment (silabs-hfegran) [reply cv32e40s/env/corev-dv/cv32e40s_zcmt_instr_lib.sv:153]: yes, they are defined as rs1 in riscv-dv if I recall correctly

**Pull request merged**
